clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -580,6 +583,49 @@ public void testConnectDisconnectDuringInSinglePoll() throws Exception {
         control.verify();
     }
 
+    @Test
+    public void testOutboundConnectionsCountInConnectionCreationMetric() throws Exception {
+        // create connections
+        int conns = 5;
+        InetSocketAddress addr = new InetSocketAddress("localhost", server.port);
+        for (int i = 0; i < conns; i++)
+            connect(Integer.toString(i), addr);
+
+        // Poll continuously, as we cannot guarantee that the first call will see all connections
+        for (int i = 0; i < 10; i++) {
+            selector.poll(0L);
+            if (selector.connected().size() == conns)
```

Comment:
`selector.connected()` is cleared after each `poll()`. So we should keep track of the total number and compare the total against `conns`.

Reply:
Absolutely true, good catch!